server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/TrustingSslSocketFactory.java

```diff
@@ -39,10 +41,17 @@
  */
 public class TrustingSslSocketFactory extends SocketFactory {
 
+    private static final Lock mutex = new ReentrantLock();
+
     private final SSLSocketFactory delegate;
 
-    public static synchronized SocketFactory getDefault() {
-        return new TrustingSslSocketFactory();
+    public static SocketFactory getDefault() {
```

Comment:
any idea why this method was synchronized?

Reply:
> any idea why this method was synchronized?

I vaguely remember having issues with some SslSocketFactories that fail in constructors when created concurrently.  I don't think that we have any tests that use LDAP over SSL, I'll try to test it manually -- see if maybe we can remote the synchronization.